libs/virtual-node/virtual-node-datamodel/src/main/kotlin/net/corda/libs/virtualnode/datamodel/VirtualNodeEntity.kt

```diff
@@ -124,8 +125,12 @@ fun EntityManager.findVirtualNode(holdingIdentityShortHash: String): VirtualNode
         queryBuilder
     }
 
-    return createQuery(queryBuilder)
-        .setParameter("shortId", holdingIdentityShortHash)
-        .setMaxResults(1)
-        .singleResult
+    return try {
+        createQuery(queryBuilder)
+            .setParameter("shortId", holdingIdentityShortHash)
+            .setMaxResults(1)
+            .singleResult
```

Comment:
Or we can even do:
```suggestion
            .resultList.singleOrNull
```
to avoid handling the exception thrown.